### PR TITLE
3.3.0 Feature Major System

### LINKED ISSUE #1646: Major-System Code
GC78AYH

### PR DESCRIPTION
**Pull request by olliklee**
Got this to work for DE, EN, FR and PL.
It's now class based.
Closes #1646

### CODE REVIEW

File: lib/tools/crypto_and_encodings/major_system/logic/major_system.dart [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+    if (nounMode) {
+      return words.where((word) => isUpperCase(word[0]))
+          .join(' ')
+          .toLowerCase();
+    }
+    return words.join(' ').toLowerCase();
+  }
+
+// returns a list of consonant groups like [d, r, m, nd, g, ht, m, s, chs]
```
Comment (Mike-3) [inline lib/tools/crypto_and_encodings/major_system/logic/major_system.dart:49]: with 3 slashes

File: lib/application/searchstrings/assets/en.json [hunk 2]
```diff
@@ -274,6 +274,7 @@
   "lcm": "lcm leastcommonmultiplier",
   "licenses": "licenses licences library libraries",
   "magic_eye": "3dstereograms autostereograms 3dpictures 3dimages",
+  "major_system": "phoneticnnumbersystem phoneticmnemonicsystem herigonesmnemonicsystem",
```
Comment (Mike-3) [inline lib/application/searchstrings/assets/en.json:277]: phonetic is double

File: lib/tools/crypto_and_encodings/major_system/logic/major_system.dart [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+  String _translateGroup(String group) {
+    group = _replaceDoubleLetters(group);
+    _specialTranslations.forEach((pattern, replacement) {
+      group = group.replaceAll(pattern, replacement);
+    });
+    return group.split('').map((char) => _translations[char] ?? '').join();
+  }
+
+  String _replaceDoubleLetters(String input) {
```
Comment (Mike-3) [inline lib/tools/crypto_and_encodings/major_system/logic/major_system.dart:66]: please use removeDuplicateCharacters  if possible

File: lib/tools/crypto_and_encodings/major_system/widget/major_system.dart [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+            },
+            items: MajorSystemLanguage.values.map((language) {
+              return GCWDropDownMenuItem(
+                  value: language,
+                  child: i18n(context, languageName(language)));
+            }).toList(),
+        ),
+        GCWTextDivider(
+            text: i18n(context, 'major_system_settings_capitalized_only')),
+        GCWTwoOptionsSwitch(
```
Comment (Mike-3) [inline lib/tools/crypto_and_encodings/major_system/widget/major_system.dart:68]: image
The switch confused me.
Can you please just put in a GCWOnOffSwitch with 'Noun Mode'.

**Review by Mike-3 — CHANGES_REQUESTED** [review]

File: lib/tools/crypto_and_encodings/major_system/logic/major_system.dart (hunk 3 above)
Comment (olliklee) [reply lib/tools/crypto_and_encodings/major_system/logic/major_system.dart:66]: It's not the same.

File: lib/tools/crypto_and_encodings/major_system/widget/major_system.dart (hunk 4 above)
Comment (olliklee) [reply lib/tools/crypto_and_encodings/major_system/widget/major_system.dart]: Yes sure. Much better.

**Pull request merged**